src/connector/src/parser/postgres.rs

```diff
@@ -418,7 +420,8 @@ fn pg_numeric_to_varchar(val: Option<PgNumeric>) -> Option<ScalarImpl> {
 
 fn pg_numeric_to_string(val: Option<PgNumeric>) -> Option<String> {
     if let Some(pg_numeric) = val {
-        // TODO(kexiang): NEGATIVE_INFINITY -> -Infinity, POSITIVE_INFINITY -> Infinity, NAN -> NaN
+        // FIXME(kexiang): NEGATIVE_INFINITY -> -Infinity, POSITIVE_INFINITY -> Infinity, NAN -> NaN
+        // https://github.com/risingwavelabs/risingwave/issues/16395
         // The current implementation is to ensure consistency with the behavior of cdc event parsor.
         match pg_numeric {
             PgNumeric::NegativeInf => Some(String::from("NEGATIVE_INFINITY")),
```

Comment:
May I ask why this is not trivial to fix? The issue and the comment it linked to seems not so detailed and I still don't quite get it...

Reply:
https://github.com/risingwavelabs/risingwave/pull/16346#discussion_r1571704867

Here, we choose to output as these values is simply to keep consistent with json parser's behavior.

In json parser, we currently have no pg-type information. The only thing we know is that this cell is currently a string (a string in a json) and we want to convert it to varchar. When encountering a "NEGATIVE_INFINITY", we cannot simply map it to "Infinity", as the datum may belong to a varchar column in PG and "NEGATIVE_INFINITY" may be original value. Otherwise, the user's data may be modified. 

To solve this issue, we need to pass pg types information as parameter into the json parser, which is non-trivial.

Reply:
> Here, we choose to output as these values is simply to keep consistent with json parser's behavior.

PTAL https://github.com/risingwavelabs/risingwave/issues/16395#issuecomment-2084532557

Reply:
I'll take a try to check whether `precise` can help produce correct strings.